pkg/packagekit/updates.jsx

```diff
@@ -682,15 +692,29 @@ class OsUpdates extends React.Component {
     initialLoadOrRefresh() {
         PK.watchRedHatSubscription(registered => this.setState({ unregistered: !registered }));
 
+        cockpit.addEventListener("visibilitychange", () => {
+            if (!cockpit.hidden)
+                this.loadOrRefresh(false);
+        });
+
+        if (!cockpit.hidden)
+            this.loadOrRefresh(true);
+        else
+            this.loadUpdates();
+    }
+
+    loadOrRefresh(always_load) {
         PK.call("/org/freedesktop/PackageKit", "org.freedesktop.PackageKit", "GetTimeSinceAction",
                 [PK.Enum.ROLE_REFRESH_CACHE])
-                .done(seconds => {
+                .done(results => {
+                    const seconds = results[0];
```

Comment:
If `results` is  a list here, I'm a bit stumped how it worked before? It seems to me that the "refresh indexes if they are older than a day" logic actually worked in practice.

Reply:
Yes, it did work, because of JS magic. `[ 123 ] < 124` => `true`.  I guess this should be a separate commit.

Reply:
OMGponies -- thanks for the explanation!